File: c/array/search-insert-position.c
int searchInsert(int* nums, int numsSize, int target) {
    int start,end,mid;

    start = 0;
    end = numsSize - 1;
    while(start < end){
        mid = (start + end) / 2;
        if(nums[mid] == target)
            return mid;
        else if(nums[mid] > target)
            end = mid - 1;
        else
            start = mid + 1;
    }

    return nums[start] < target ? start + 1 : start;
}

File: go/array/search-insert-position.go
func searchInsert(nums []int, target int) int {
    i, j := 0, len(nums) - 1
    for i <= j {
        mid := (i + j) >> 1
        if nums[mid] == target {
            return mid
        }else if nums[mid] > target {
            j = mid - 1
        }else {
            i = mid + 1
        }
    }

    return i
}
